### PR TITLE
Enabling Windows development

### PR DESCRIPTION
**Pull request by Bedzior**
- [x] Windows target builds using the default generator (assuming MSVC)
- [x] `.\scripts\dependencies-install.ps1` downloads, installs SDL from [libsdl.org](https://www.libsdl.org/)
- [x] Provided `.\scripts\config-windows.ps1` and `.\scripts\build-windows.ps1`, analogous to PSP and Linux ones
- [x] `SDL.dll` is placed into build and install directories
- [x] Resolved Desktop similarities by renaming Linux-only classes to Desktop-related, unified some behaviour in CMake files
- [x] Included a Travis build, using the provided scripts via experimental `powershell` support
- [x] Deploying `SpelunkyWindows.zip` on tag (contains executable and `SDL.dll` for now)

- [x] I also extended README with Windows-related info

TODO:
- [ ] Fixing Timestep (std::chrono on Windows could resolve the framerate issue)

Rafal

### CODE REVIEW

**Comment by dbeef** [comment]
Good job, I am looking forward to this.
Please update readme section on building; split it into each platform, add information on the (so far) only external dependency which is SDL 1.xx

File: scripts/setup-windows.ps1 [hunk 1]
```diff
@@ -0,0 +1,32 @@
+[CmdletBinding()]
```
Comment (dbeef) [inline scripts/setup-windows.ps1:1]: I am not entirely for the idea of scripts for dependency installation.

Please rename this file to `dependencies-install.ps1`

File: src/time/CMakeLists.txt [hunk 2]
```diff
@@ -6,6 +6,7 @@ add_library(Time STATIC
 )
 
 target_include_directories(Time
+        PUBLIC ${SDL_INCLUDE_DIR}
```
Comment (dbeef) [inline src/time/CMakeLists.txt:9]: Please create INTERFACE target carrying `SDL_INCLUDE_DIR` and `SDL_LIBRARY` for convenience.
Call it SDL_1_XX.

File: scripts/setup-windows.ps1 (hunk 1 above)
Comment (Bedzior) [reply scripts/setup-windows.ps1:1]: I wonder, where did you get the script's name from? 🤔

**Comment by Bedzior** [comment]
I created an explicit SDL_1_XX target to handle SDL dependency in `Input` and `Context` modules, to appease GCC's `dl` linking order.

**Pull request merged**